File: examples/python-versions/planet_sort.py
class Planet:
    def __init__(self, name, a, e):
        self.name = name
        self.a = a
        self.e = e

    def __str__(self):
        ostr = f"{self.name:12} : ({self.a:8}, {self.e:8})"
        return ostr


def main():

    planets = [Planet("Mercury",  0.3871, 0.2056),
               Planet("Venus",    0.7233, 0.0068),
               Planet("Earth",    1.0000, 0.0167),
               Planet("Mars",     1.5237, 0.0934),
               Planet("Jupiter",  5.2029, 0.0484),
               Planet("Saturn",   9.5370, 0.0539),
               Planet("Uranus",  19.189,  0.0473),
               Planet("Neptune", 30.070,  0.0086)]

    planets.sort(key=lambda p: p.e)

    for p in planets:
        print(p)


if __name__ == "__main__":
    main()

File: examples/functions/planet_sort.cpp
#include <iostream>
#include <iomanip>
#include <format>
#include <vector>
#include <algorithm>

struct Planet
{
    std::string name{};
    double a{};            // semi-major axis
    double e{};            // eccentricity
};

std::ostream& operator<< (std::ostream& os, const Planet& p) {

    os << std::format("{:12} : ({:8}, {:8}))", p.name, p.a, p.e);
    return os;
}

int main() {

    std::vector<Planet> planets{{.name="Mercury", .a=0.3871, .e=0.2056},
                                {.name="Venus",   .a=0.7233, .e=0.0068},
                                {.name="Earth",   .a=1.0000, .e=0.0167},
                                {.name="Mars",    .a=1.5237, .e=0.0934},
                                {.name="Jupiter", .a=5.2029, .e=0.0484},
                                {.name="Saturn",  .a=9.5370, .e=0.0539},
                                {.name="Uranus",  .a=19.189, .e=0.0473},
                                {.name="Neptune", .a=30.070, .e=0.0086}};

    std::sort(planets.begin(), planets.end(),
              [] (const Planet& a, const Planet& b) {return a.e < b.e;});

    for (const auto& p : planets) {
        std::cout << p << std::endl;
    }

}
